### ISSUE TITLE
CMF SageMath code

### ISSUE THREAD

**Issue by jvoight** [LMFDB/lmfdb#5396]
The "Code to Sage" link for http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/23/2/a/a/ which links to
http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/download_code_newform/23.2.a.a/sage
gives code that doesn't work.  

```
sage: # SageMath code for working with modular form 23.2.a.a
....: 
....: 
....: # Compute space of new eigenforms: 
....: D = DirichletGroup(23)
....: c = D.Element(D,vector(None))
....: N = Newforms(c,2,names="a")
....: 
....: # select newform: 
....: f = N[1]
....: 
....: # q-expansion: 
....: f.q_expansion() # note that sage often uses an isomorphic number field
....: 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-cb625641ff18> in <module>()
      4 # Compute space of new eigenforms:
      5 D = DirichletGroup(Integer(23))
----> 6 c = D.Element(D,vector(None))
      7 N = Newforms(c,Integer(2),names="a")
      8 

/usr/local/SageMath/local/lib/python2.7/site-packages/sage/modules/free_module_element.pyx in sage.modules.free_module_element.vector (build/cythonized/sage/modules/free_module_element.c:5992)()
    510             sparse = False
    511 
--> 512     v, R = prepare(v, R, degree)
    513 
    514     if sparse:

/usr/local/SageMath/local/lib/python2.7/site-packages/sage/modules/free_module_element.pyx in sage.modules.free_module_element.prepare (build/cythonized/sage/modules/free_module_element.c:6541)()
    609         except TypeError:
    610             pass
--> 611     v = Sequence(v, universe=R, use_sage_types=True)
    612     ring = v.universe()
    613     if not is_Ring(ring):

/usr/local/SageMath/local/lib/python2.7/site-packages/sage/structure/sequence.pyc in Sequence(x, universe, check, immutable, cr, cr_str, use_sage_types)
    231     if universe is None:
    232         orig_x = x
--> 233         x = list(x) # make a copy even if x is a list, we're going to change it
    234 
    235         if len(x) == 0:

TypeError: 'NoneType' object is not iterable
```

I think two fixes are required:

```
# SageMath code for working with modular form 23.2.a.a


# Compute space of new eigenforms:
D = DirichletGroup(23)
c = D.Element(D,[1])
N = Newforms(c,2,names="a")

# select newform:
f = N[0]

# q-expansion:
f.q_expansion() # note that sage often uses an isomorphic number field
```

**Comment by pascalmolin** [comment]
My fault, the sage and magma code are wrong indeed, inherited from the old mf code.
Can someone provide a quickfix ? otherwise I comment it.

**Comment by edgarcosta** [comment]
I will try to fix the snippet tomorrow, but I am unsure about their value as a download.

**Comment by AndrewVSutherland** [comment]
Fixed by #5402 (which I will merge as soon as tests finish).  I also created a separate issue to address the fact that even with a working snippet, it likely won't give you the newform you want.
